src/libs/relay/conduit_relay_io_blueprint.cpp

```diff
@@ -0,0 +1,243 @@
+// 
+// For details, see: http://software.llnl.gov/conduit/.
+// 
+// Please also read conduit/LICENSE
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the disclaimer below.
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+// 
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, 
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+// POSSIBILITY OF SUCH DAMAGE.
+// 
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+//-----------------------------------------------------------------------------
+///
+/// file: conduit_relay_io_blueprint.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include "conduit_relay_io_blueprint.hpp"
+#include "conduit_relay_io.hpp"
+
+#include "conduit_blueprint.hpp"
+
+//-----------------------------------------------------------------------------
+// standard lib includes
+//-----------------------------------------------------------------------------
+#include <iostream>
+
+
+//-----------------------------------------------------------------------------
+// -- begin conduit:: --
+//-----------------------------------------------------------------------------
+namespace conduit
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay --
+//-----------------------------------------------------------------------------
+namespace relay
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay::io_blueprint --
+//-----------------------------------------------------------------------------
+namespace io_blueprint
+{
```

Comment:
header seems to say `blueprintio`, and this file `io_blueprint`?

I woudl actually go with a nested name space:
`io::blueprint`

Reply:
@cyrush: I actually had that as the name originally, but I changed the namespace inclusion to an underscore to prevent potential ambiguity when `using namespace` directives get involved in the mix. Do you still prefer the nested namespace idea despite this potential problem? If so, I don't mind changing it to suit your suggestion.

Reply:
@cyrus: After giving the rename a go, I really don't think it's a good idea because the potential ambiguities make writing the Blueprint/Relay code nontrivially more involved. For example, just trying the rename on its own resulted in the following errors:
```
/path/to/conduit/src/libs/relay/conduit_relay_io_blueprint.cpp: In function 'void conduit::relay::io::blueprint::save(const conduit::Node&, const string&, const string&)':
/path/to/conduit/src/libs/relay/conduit_relay_io_blueprint.cpp:148:20: error: 'conduit::relay::io::blueprint::mesh' has not been declared
     if(!blueprint::mesh::verify(mesh, info))
                    ^
/path/to/conduit/src/libs/relay/conduit_relay_io_blueprint.cpp:158:19: error: 'conduit::relay::io::blueprint::mesh' has not been declared
     if(blueprint::mesh::is_multi_domain(mesh))
                   ^
/path/to/conduit/src/libs/relay/conduit_relay_io_blueprint.cpp:184:33: error: 'conduit::relay::io::blueprint::mesh' has not been declared
                     !blueprint::mesh::topology::unstructured::verify(topo, info) ||
                                 ^
/path/to/conduit/src/libs/relay/conduit_relay_io_blueprint.cpp:190:28: error: 'conduit::relay::io::blueprint::mesh' has not been declared
                 blueprint::mesh::generate_index(
```
With this in mind, I'm going to keep it as `io_blueprint` for the time being.

Reply:
gotcha, that sounds like a good plan for now.

I have some thoughts about how to better centralize some of the mesh related relay read and write stuff that 
we can pursue after this PR. 

Working on getting ADIOS support in, and with that will have a mpi enabled relay io library.

That library will be a good place to centralize similar logic in VisIt, Ascent, and Conduit.